### ISSUE TITLE
<amp-video-iframe> autoplay doesn't work

### ISSUE THREAD

**Issue by skyzanellap** [ampproject/amphtml#25400]
Hi everyone,
I'm trying to set video autoplay using amp-video-iframe tag but I noticed that not works properly.
Example: http://videoplatform.sky.it/test/support/amp/options/1/amp.html
Our external player is based on JWplayer and our customization.

To reproduce the issue:
1. Open the link in browser
2. Player not works, it's always in buffering state
2. Open developer tool (such as Chrome Developer Tools)
3. You can see this error: **Uncaught TypeError: Cannot read property 'on' of null**

Why? Can you help us? Any idea?

All browsers are affected (Chrome, Firefox and Safari mobile) and we are using this AMP version: 1910302211460.
Thanks!
Pietro

**Comment by gilbertococchi** [comment]
@nainar @alanorozco can you please check this test page?

From what I can see they are correctly mapping the VIDEO.jw-video Element here:

`
player.on("ready", () => {				
				(window.AmpVideoIframe = window.AmpVideoIframe || []).push(function(ampIntegration) {
					const videoTag = document.querySelector('.jw-video');
					ampIntegration.listenTo('jwplayer', videoTag);
				});
			});
`

but I don't know the hidden details of JW to be sure the VIDEO is mapped correctly.

Also it seems like there are other VIDEO objects for Advertising, shoudl they be mapped as well?

**Comment by nainar** [comment]
`ampIntegration.listenTo('jwplayer', videoTag)` might be the problem are we sure `videoTag` is a JWPlayer instance?

`ampIntegration.listenTo('jwplayer', player.getJwPlayer())` <- potential fix.

**Comment by gilbertococchi** [comment]
Thanks Naina, @skyzanellap I've tried to apply what Naina suggested by using Chrome Local Overrides (really useful to test something on a site without pushing code changes).

<img width="626" alt="Screenshot 2019-11-11 at 18 50 27" src="https://user-images.githubusercontent.com/329023/68608910-55e64480-04b4-11ea-90c3-adef45fb2760.png">

I cannot see the error anymore but the player is not loading or playing the video either.

<img width="1419" alt="Screenshot 2019-11-11 at 18 53 50" src="https://user-images.githubusercontent.com/329023/68609046-9e9dfd80-04b4-11ea-8048-727383239882.png">

I can see the Player is passed but it doesn't allow to reproduce the video.

**Comment by nainar** [comment]
@alanorozco and @wassgha if you could take a look please?

**Comment by wassgha** [comment]
I don't think locally overriding the script is a good way to test this (so many moving parts), let's wait to hear back from @skyzanellap once the change is implemented on the server

**Comment by gilbertococchi** [comment]
@skyzanellap can you please upload on the external Iframe URL the code changes to test this again?

**Comment by gilbertococchi** [comment]
@wassgha @nainar can you please take a look at this page again?
http://videoplatform.sky.it//test/support/amp/options/1/index.html

@skyzanellap updated the code with the Player object but the Player stays pending without loading the actual video nor triggering any errors...

**Comment by alanorozco** [comment]
@gilbertococchi 

I'm wondering if this follows Jwplayer contract? Is the `JwPlayer`'s `on()` method wrapped somehow? From trying to use it manually, it does nothing. But using the outer `player` does work.

Otherwise you could integrate manually (e.g. without `listenTo()`).

**Comment by gilbertococchi** [comment]
Hi Alan, I've tried to edit Sky's example locally but it never works, can you please clarify how it would be possible to integrated without listenTo?
Do you have in mind some examples of live integrations with amp-video-iframe on jwplayer working?

If I send the player object with the on() method nothing happens, If I return the Video player lots of JS errors on missing controls would trigger.

**Comment by alanorozco** [comment]
@gilbertococchi 

Sure: [amp-video-iframe-jwplayer.glitch.me](https://amp-video-iframe-jwplayer.glitch.me/)

You can see examples that either use `listenTo` or not. Both integrate with Jwplayer.

For more info on avoiding `listenTo` with custom configurations like these, see [example code](https://glitch.com/edit/#!/amp-video-iframe-jwplayer?path=jwplayer-no-listento.html) and [`postEvent` and `method` in docs](https://amp.dev/documentation/components/amp-video-iframe/#integration-inside-the-frame)

**Comment by alanorozco** [comment]
@gilbertococchi 

I was also able to [integrate with the custom player](https://glitch.com/edit/#!/amp-video-iframe-jwplayer?path=skyplayer-no-listento.html)

Findings:

1. You should **not** `autostart`. Let amp-video-iframe send playback signals for autoplay:

```js
const params = {
  //  ...
  autostart: false,
};
```

2. As suspected, `on()` needs to be called on the outer `SkyPlayer`, while playback methods need to be called on the inner `Jwplayer`, like:

```js
player.on("error", () => amp.postEvent("error"));
// vs
amp.method("pause", () => player.getJwPlayer().pause());
```

4. There's some weirdness that expects all players to be paused first:

```js
amp.method("play", () => {
  SkyPlayerManager.pauseAllPlayers(container);
  player.getJwPlayer().play();
});
```

See [full code](https://glitch.com/edit/#!/amp-video-iframe-jwplayer?path=skyplayer-no-listento.html).

Closing, since this is an API usage issue.
